api/control/v1alpha1/types.go

```diff
@@ -128,28 +119,52 @@ const (
 // and modulates how they are downsynced.
-// One modulation is specifying a set of StatusCollectors to apply
-// to returned status.
-// The other modulation is specifying whether the propagation from WDS to WEC
-// involves continual maintenance of the spec or the object is only created
-// if it is absent.
 type DownsyncPolicyClause struct {
 	DownsyncObjectTest `json:",inline"`
+	DownsyncModulation `json:",inline"`
+}
 
+// DownsyncModulation is about variations on downsync behavior.
+type DownsyncModulation struct {
 	// `createOnly` indicates that in a given WEC, the object is not to be updated
 	// if it already exists.
 	// +optional
 	CreateOnly bool `json:"createOnly,omitempty"`
 
-	// `statusCollection` holds the rules for collecting status from the WECs.
+	// `statusCollectors` is a list of references of StatusCollectors to apply.
 	// +optional
-	StatusCollection *StatusCollection `json:"statusCollection,omitempty"`
-}
-
-// StatusCollection holds the rules for collecting status from the WECs.
-type StatusCollection struct {
-	// `statusCollectors` is a list of StatusCollectors name references to be applied.
 	StatusCollectors []string `json:"statusCollectors,omitempty"`
+
+	// WantSingletonReportedState, in short, indicates an expectation
+	// that the matching workload objects are distributed to exactly one WEC
+	// and requests that the `.status` of such objects propagate from the WEC
+	// to the WDS.
+	//
+	// For a precise description, start with a couple of definitions.
+	// For a given workload object, _singleton status return is requested_
+	// if and only if there exists at least one BindingPolicy or Binding
+	// that has `wantSingletonReportedState==true` in a clause that
+	// matches/references the workload object.
+	//
+	// The _qualified WEC set_ of a workload object is the set of WECs that are
+	// associated with that workload object by at least one BindingPolicy or Binding
+	// that has `wantSingletonReportedState==true` in a clause that
+	// matches/references the workload object.
+	//
+	// For a given workload object, while singleton status return is requested,
+	// KubeStellar maintains a label on the object whose name (key) is
+	// `kubestellar.io/executing-count` and whose value is a string representation
+	// of the size of the qualified WEC set of that object.
+	// While singleton status return is _not_ requested, KubeStellar suppresses
+	// the existence of a label with that name (key).
+	// While singleton status return is requested _and_ the size of the qualified
+	// WEC set is 1, KubeStellar propagates the object's `.status` from that WEC
+	// to the `.status` section of the object in the WDS.
+	// While either singleton status return is NOT requested or the size of the
+	// qualified WEC set is NOT 1, the `.status` of the object in the WDS does
+	// not have contents propagated there from a WEC by KubeStellar.
+	// +optional
+	WantSingletonReportedState bool `json:"wantSingletonReportedState,omitempty"`
```

Comment:
This should not be the only place this API is explained thoroughly, the docs should reflect it as well.

Reply:
I have revised the PR, clarifying the last sentence and adding doc updates.